File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in range(len(s)):
            if s[i] in "([{":
                stack.append(s[i])
            elif stack==[] or (stack[-1] == "(" and s[i]!=")") or (stack[-1] == "{" and s[i]!="}") or (stack[-1] == "[" and s[i]!="]"):
                return False
            else:
                stack.pop()
                
        return stack == []

File: Data Structures & Algorithms/daily-temperatures/submission-3.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        List<Integer> stack = new ArrayList<>();
        int[] sol = new int[temperatures.length];

        for (int i = temperatures.length - 1; i >= 0; i--){
            
            while (stack.size() != 0 && temperatures[stack.get(stack.size()-1)] <= temperatures[i]){
                stack.remove(stack.size()-1);
            }

            if (stack.size() == 0){
                sol[i] = 0;
            } else {
                sol[i] = stack.get(stack.size()-1) - i;
            }

            stack.add(i);

        }

        return sol; 

    }
}
